### PR TITLE
ticdc: refactoring the document structure about ticdc

### PR DESCRIPTION
**Pull request by leoppro**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: ticdc/ticdc-changefeed-overview.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+- ⑥ `changefeed` 已经进行到预设的 TargetTs，同步自动停止。
+- ⑦ `changefeed` 停滞时间超过 `gc-ttl` 所指定的时长，不可被恢复。
+- ⑧ `changefeed` 尝试自动恢复过程中发生不可恢复的错误。
+
+## 操作 Changefeed
+
+本文档介绍如何通过 TiCDC 提供的命令行工具 `cdc cli` 管理 TiCDC 集群和同步任务，并介绍了如何使用 TiUP 来升级和修改 TiCDC 集群的配置。你也可以通过 HTTP 接口，即 TiCDC OpenAPI 来管理 TiCDC 集群和同步任务，详见 [TiCDC OpenAPI](/ticdc/ticdc-open-api.md)。
+
+
+本部分介绍如何使用 TiCDC 命令行工具来管理集群状态。如果你使用的 TiCDC 是用 TiUP 部署的，可以通过 `tiup ctl:<version> cdc` 来使用 TiCDC 命令行工具，注意需要将 `<version>` 替换为 TiCDC 集群版本。你也可以通过直接执行 `cdc cli` 直接使用命令行工具。
```
Comment (IANTHEREAL) [inline ticdc/ticdc-changefeed-overview.md:40]:
```suggestion
如果你使用的 TiCDC 是用 TiUP 部署的，可以通过 `tiup ctl:<version> cdc` 来使用 TiCDC 命令行工具，注意需要将 `<version>` 替换为 TiCDC 集群版本。你也可以通过直接执行 `cdc cli` 直接使用命令行工具。
```

File: ticdc/deploy-ticdc.md [hunk 2]
```diff
@@ -24,56 +24,138 @@ summary: 了解 TiCDC 软硬件环境要求以及如何安装部署 TiCDC。
...
+扩容的方式与部署 TiCDC 集群的方式类似，推荐使用 TiUP 工具完成。 首先，编写一个名称为scale-out.yaml 的配置文件，包含需要扩容的节点的配置信息，下面是一个示例：
+
+```
+cdc_servers:
+  - host: 10.1.1.1
+    gc-ttl: 86400
+    data_dir: /data/deploy/install/data/cdc-8300
+  - host: 10.1.1.2
+    gc-ttl: 86400
+    data_dir: /data/deploy/install/data/cdc-8300
```
Comment (IANTHEREAL) [inline ticdc/deploy-ticdc.md:52]:
```suggestion
    data_dir: /data/deploy/install/data/cdc-8300
 - host: 10.0.1.4:8300
    gc-ttl: 86400
    data_dir: /data/deploy/install/data/cdc-8300
```

File: ticdc/deploy-ticdc.md [hunk 3]
```diff
@@ -24,56 +24,138 @@ summary: 了解 TiCDC 软硬件环境要求以及如何安装部署 TiCDC。
...
+ - 终止 TiCDC 节点：`tiup cluster stop -R cdc`
+ - 启动 TiCDC 节点：`tiup cluster start -R cdc`
+ - 重启 TiCDC 节点：`tiup cluster restart -R cdc`
 
-* 集群中至少有两个正在运行的 TiCDC 实例。
-* TiUP 版本至少为 v1.11.0。
+## 使用加密传输 (TLS) 功能
 
-满足上述条件后，即可执行 `tiup cluster upgrade` 命令对集群进行滚动升级：
+请参阅[为 TiDB 组件间通信开启加密传输](/enable-tls-between-components.md)。
```
Comment (IANTHEREAL) [inline ticdc/deploy-ticdc.md:132]: 这里不是很明白，如何通过 TIUP 来开启，我在这个文档没找到  https://github.com/pingcap/docs-cn/blob/dacaa863960db3c4e92c88e38162c4781f45f882/tiup/tiup-cluster-topology-reference.md#cdc_servers

File: ticdc/ticdc-changefeed-overview.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
+---
+title: Changefeed 概述
+summary: 了解 Changefeed 的基本概念和 Changefeed 状态的定义与流转
+---
+
+# Changefeed 概述
```
Comment (IANTHEREAL) [inline ticdc/ticdc-changefeed-overview.md:6]: 这个文档不像是个 overview，像是介绍运行原理的。 anyway 先这样

File: ticdc/ticdc-overview.md [hunk 5]
```diff
@@ -8,54 +8,33 @@ aliases: ['/docs-cn/dev/ticdc/ticdc-overview/','/docs-cn/dev/reference/tools/tic
...
-- `capture`：TiCDC 运行进程，多个 `capture` 组成一个 TiCDC 集群，负责 KV change log 的同步。
-    - 每个 `capture` 负责拉取一部分 KV change log。
-    - 对拉取的一个或多个 KV change log 进行排序。
-    - 向下游还原事务或按照 TiCDC Open Protocol 进行输出。
-
-## 同步功能介绍
+### 核心能力
 
-本部分介绍 TiCDC 的同步功能。
+- 提供 TiDB -> TiDB/MySQL（或其他兼容 MySQL 协议的数据库）的低延迟增量数据同步能力
```
Comment (IANTHEREAL) [inline ticdc/ticdc-overview.md:19]:
```suggestion
- 提供 TiDB -> TiDB 之间数据容灾复制的能力，实现秒级别 RPO 和分钟级别 RTO
- 提供 TiDB -> MySQL（或其他兼容 MySQL 协议的数据库）的低延迟的增量数据同步能力
```

File: ticdc/ticdc-overview.md [hunk 6]
```diff
@@ -8,54 +8,33 @@ aliases: ['/docs-cn/dev/ticdc/ticdc-overview/','/docs-cn/dev/reference/tools/tic
...
+- 提供 TiDB -> Kafka 增量数据同步能力，推荐的数据格式包含 [Canal-JSON](/ticdc/ticdc-canal-json.md)，[Avro](/ticdc/ticdc-avro-protocol.md) 等
+- 提供表级别数据同步能力，支持同步过程中过滤数据库、表的能力
+- 数据同步支持最终一致性
```
Comment (IANTHEREAL) [inline ticdc/ticdc-overview.md:22]:
```suggestion
```

File: ticdc/ticdc-overview.md [hunk 7]
```diff
@@ -8,54 +8,33 @@ aliases: ['/docs-cn/dev/ticdc/ticdc-overview/','/docs-cn/dev/reference/tools/tic
...
+- 高可用架构，无单点故障；支持用户动态添加、删除 TiCDC 节点
+- 支持通过 [Open API](/ticdc/ticdc-open-api.md) 进行集群管理，包括查询任务状态；动态修改任务配置；动态创建、删除任务等
 
-### sink 支持
-
-目前 TiCDC sink 模块支持同步数据到以下下游：
-
-- MySQL 协议兼容的数据库，提供最终一致性支持。
-- 以 TiCDC Open Protocol 输出到 Kafka，可实现行级别有序、最终一致性或严格事务一致性三种一致性保证。
+要快速了解 TiCDC 的基本原理和使用方法，建议先观看下面的培训视频（时长 33 分钟）。注意本视频只为学习参考，具体操作步骤和最新功能，请以文档内容为准。
```
Comment (IANTHEREAL) [inline ticdc/ticdc-overview.md:26]: 这个视频是什么版本的，是否适合 6.5 及之后版本？

File: ticdc/ticdc-overview.md [hunk 8]
```diff
@@ -8,54 +8,33 @@ aliases: ['/docs-cn/dev/ticdc/ticdc-overview/','/docs-cn/dev/reference/tools/tic
...
-### 同步顺序保证和一致性保证
+<video src="https://download.pingcap.com/docs-cn%2FLesson22_ticdc.mp4" width="100%" height="100%" controls="controls" poster="https://tidb-docs.s3.us-east-2.amazonaws.com/thumbnail+-+lesson+22.png"></video>
 
-#### 数据同步顺序
+### 数据同步顺序性
 
 - TiCDC 对于所有的 DDL/DML 都能对外输出**至少一次**。
 - TiCDC 在 TiKV/TiCDC 集群故障期间可能会重复发相同的 DDL/DML。对于重复的 DDL/DML：
     - MySQL sink 可以重复执行 DDL，对于在下游可重入的 DDL （譬如 truncate table）直接执行成功；对于在下游不可重入的 DDL（譬如 create table），执行失败，TiCDC 会忽略错误继续同步。
     - Kafka sink 会发送重复的消息，但重复消息不会破坏 Resolved Ts 的约束，用户可以在 Kafka 消费端进行过滤。
```
Comment (IANTHEREAL) [inline ticdc/ticdc-overview.md:35]:
```suggestion
    - Kafka sink 会发送重复的消息，但重复消息不会破坏 checkpoint 的约束，用户可以在 Kafka 消费端进行过滤。
```

File: ticdc/ticdc-overview.md [hunk 9]
```diff
@@ -72,9 +50,22 @@ TiCDC 的系统架构如下图所示：
 
     - TiCDC 提供不同的数据分发策略，可以按照表、主键或 ts 等策略分发数据到不同 Kafka partition。
     - 不同分发策略下 consumer 的不同实现方式，可以实现不同级别的一致性，包括行级别有序、最终一致性或跨表事务一致性。
```
Comment (IANTHEREAL) [inline ticdc/ticdc-overview.md:52]: 这里没描述的太清楚，后面也没见到介绍如何实现行级别有序，此外对于 kafka sink 没有什么最终一致性或者跨表事物一致性的概念吧，都统一是 checkpoint 水位线控制，描述的太复杂了，而且不知道怎么用

File: ticdc/ticdc-overview.md [hunk 10]
```diff
@@ -72,9 +50,22 @@ TiCDC 的系统架构如下图所示：
...
-    - TiCDC 没有提供 Kafka 消费端实现，只提供了 [TiCDC 开放数据协议](/ticdc/ticdc-open-protocol.md)，用户可以依据该协议实现 Kafka 数据的消费端。
+    - TiCDC 没有提供 Kafka 消费端实现，只提供了 [TiCDC 开放数据协议](/ticdc/ticdc-avro-protocol.md)，用户可以依据该协议实现 Kafka 数据的消费端。
```
Comment (IANTHEREAL) [inline ticdc/ticdc-overview.md:53]:
```suggestion
```

File: ticdc/ticdc-overview.md [hunk 11]
```diff
@@ -72,9 +50,22 @@ TiCDC 的系统架构如下图所示：
...
 
-## 同步限制
+## TiCDC 架构
+
+TiCDC 作为 TiDB 的增量数据同步工具，通过 PD 内部的 etcd 实现高可用，通过多个 TiCDC 进程获取 TiKV 节点上的数据改变，在内部进行排序、合并等处理之后，通过多个同步任务(Changefeed)，同时向多个下游系统进行数据同步。
+
+![TiCDC architecture](/media/ticdc/cdc-architecture.png)
+
+在以上的架构图中：
+- TiKV ： 代表 TiDB 集群中的 TiKV 节点，当数据发生改变时 TiKV 节点会主动将发生的数据改变以变更日志（KV change logs，简称 change logs）的方式发送给 TiCDC 节点。当然，当 TiCDC 节点发现收到的 change logs 并不是连续的，也会主动发起请求，获得需要的 change logs。
```
Comment (IANTHEREAL) [inline ticdc/ticdc-overview.md:63]:
```suggestion
- TiKV Server： 代表 TiDB 集群中的 TiKV 节点，当数据发生改变时 TiKV 节点会主动将发生的数据改变以变更日志（KV change logs，简称 change logs）的方式发送给 TiCDC 节点。当然，当 TiCDC 节点发现收到的 change logs 并不是连续的，也会主动发起请求，获得需要的 change logs。 
```

File: ticdc/ticdc-overview.md [hunk 12]
```diff
@@ -72,9 +50,22 @@ TiCDC 的系统架构如下图所示：
...
-## 同步限制
+## TiCDC 架构
+
+TiCDC 作为 TiDB 的增量数据同步工具，通过 PD 内部的 etcd 实现高可用，通过多个 TiCDC 进程获取 TiKV 节点上的数据改变，在内部进行排序、合并等处理之后，通过多个同步任务(Changefeed)，同时向多个下游系统进行数据同步。
+
+![TiCDC architecture](/media/ticdc/cdc-architecture.png)
+
+在以上的架构图中：
+- TiKV ： 代表 TiDB 集群中的 TiKV 节点，当数据发生改变时 TiKV 节点会主动将发生的数据改变以变更日志（KV change logs，简称 change logs）的方式发送给 TiCDC 节点。当然，当 TiCDC 节点发现收到的 change logs 并不是连续的，也会主动发起请求，获得需要的 change logs。 
+- TiCDC cluster： 代表运行了运行 TiCDC 进程的各个节点。每个节点都运行一个 TiCDC 进程，每个进程会从 TiKV 节点中拉取一个或者多个表中的数据改变，并通过 Sink 模块同步到下游系统。
```
Comment (IANTHEREAL) [inline ticdc/ticdc-overview.md:64]:
```suggestion
- TiCDC： 代表运行了运行 TiCDC 进程的各个节点。每个节点都运行一个 TiCDC 进程，每个进程会从 TiKV 节点中拉取一个或者多个表中的数据改变，并通过 Sink 模块同步到下游系统。
```

File: ticdc/ticdc-overview.md [hunk 13]
```diff
@@ -72,9 +50,22 @@ TiCDC 的系统架构如下图所示：
...
+- PD Cluster： 代表 TiDB 集群中的调度模块，负责集群数据的事实调度，这个模块通常是由 3 个 PD 节点构成的，内部通过 etcd 集群来实现选举等高可用相关的能力。 TiCDC 集群使用了 PD 集群内置的 etcd 集群来保存自己的元数据信息，例如：节点的状态信息，changefeed 配置信息等。
```
Comment (IANTHEREAL) [inline ticdc/ticdc-overview.md:65]:
```suggestion
- PD： 代表 TiDB 集群中的调度模块，负责集群数据的事实调度，这个模块通常是由 3 个 PD 节点构成的，内部通过 etcd 集群来实现选举等高可用相关的能力。 TiCDC 集群使用了 PD 集群内置的 etcd 集群来保存自己的元数据信息，例如：节点的状态信息，changefeed 配置信息等。 
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 14]
```diff
@@ -0,0 +1,247 @@
...
+| `9092`               | 下游 Kafka 的连接端口                                          |
+| `topic-name`           | 变量，使用的 Kafka topic 名字                                      |
+| `kafka-version`      | 下游 Kafka 版本号（可选，默认值 `2.4.0`，目前支持的最低版本为 `0.11.0.2`，最高版本为 `3.2.0`。该值需要与下游 Kafka 的实际版本保持一致） |
+| `kafka-client-id`    | 指定同步任务的 Kafka 客户端的 ID（可选，默认值为 `TiCDC_sarama_producer_同步任务的 ID`） |
+| `partition-num`      | 下游 Kafka partition 数量（可选，不能大于实际 partition 数量，否则创建同步任务会失败，默认值 `3`）|
+| `max-message-bytes`  | 每次向 Kafka broker 发送消息的最大数据量（可选，默认值 `10MB`）。从 v5.0.6 和 v4.0.6 开始，默认值分别从 64MB 和 256MB 调整至 10MB。|
+| `replication-factor` | Kafka 消息保存副本数（可选，默认值 `1`）                       |
+| `protocol` | 输出到 Kafka 的消息协议，可选值有 `canal-json`、`open-protocol`、`canal`、`avro`、`maxwell` |
+| `auto-create-topic` | 当传入的 `topic-name` 在 Kafka 集群不存在时，TiCDC 是否要自动创建该 topic（可选，默认值 `true`） |
+| `enable-tidb-extension` | 可选，默认值是 `false`。当输出协议为 `canal-json` 时，如果该值为 `true`，TiCDC 会发送 Resolved 事件，并在 Kafka 消息中添加 TiDB 扩展字段。从 6.1.0 开始，该参数也可以和输出协议 `avro` 一起使用。如果该值为 `true`，TiCDC 会在 Kafka 消息中添加三个 TiDB 扩展字段。|
```
Comment (IANTHEREAL) [inline ticdc/ticdc-sink-to-kafka.md:66]: 什么扩展字段？

**Comment by IANTHEREAL** [comment]
@nongfushanquan PTAL

File: ticdc/ticdc-sink-to-mysql.md [hunk 15]
```diff
@@ -0,0 +1,124 @@
...
+| `ssl-cert`     | 连接下游 MySQL 实例所需的证书文件路径（可选） |
+| `ssl-key`      | 连接下游 MySQL 实例所需的证书密钥文件路径（可选） |
+| `time-zone`    | 连接下游 MySQL 实例时使用的时区名称，从 v4.0.8 开始生效。（可选。如果不指定该参数，使用 TiCDC 服务进程的时区；如果指定该参数但使用空值，则表示连接 MySQL 时不指定时区，使用下游默认时区） |
+| `transaction-atomicity`      | 指定事务的原子性级别（可选，默认值为 `table`）。当该值为 `table` 时 TiCDC 保证单表事务的原子性，当该值为 `none` 时 TiCDC 会拆分单表事务 |
+
+> **注意：**
+> URI 中包含特殊字符时，如 `! * ' ( ) ; : @ & = + $ , / ? % # [ ]`，需要对 URI 特殊字符进行转义处理。你可以在 [URI Encoder](https://meyerweb.com/eric/tools/dencoder/) 中对 URI 进行转义。
+
+## 灾难场景的最终一致性复制
+
```
Comment (nongfushanquan) [inline ticdc/ticdc-sink-to-mysql.md:72]: 注明该功能在 6.1.1 及以后的版本可以安全使用。

**Review by nongfushanquan — APPROVED** [review]

File: ticdc/ticdc-architecture.md [hunk 16]
```diff
@@ -0,0 +1,139 @@
...
+
+**Barrier TS**：当系统发生 DDL 语句或者使用了 TiCDC 的 sync point 时会产生的一个时间戳。
+
+  - 对于 DDL 语句，这个时间戳会用来确保在这个 DDL 语句之前的改变都被应用到下游，之后执行对应的 DDL 语句，在 DDL 语句同步完成之后再同步其他的数据改变。由于 DDL 语句的处理是 owner 角色的 Capture 负责的，DDL 语句对应的 Barrier TS 只会由 owner 节点的 processor 线程产生。
+  - sync point Barrier TS 也是一个时间戳，当用户启用 TiCDC 的 sync point 特性后，TiCDC 会根据用户指定的间隔产生一个 Barrier TS，当所有的表都同步到了这个Barrier TS 之后，记录一下对应的时间点，之后继续向下同步数据。 
+
+TiCDC 是通过对 global checkpoint TS 和 barrier TS 进行比较来确定数据是否已经同步到 barrier TS 的。如果 global checkpoint TS = barrier TS，则说明所有表都至少推进到 barrier TS 了。如果等式不成立，说明有表还没推进到该 barrier TS ，在这种情况下也不会更新 barrier TS ，因此不会有表的 sink 节点对应 resolved TS 会超过 barrier TS，即不会有表的 check point TS 超过 barrier TS。
+
+## 主要流程
+最后，对 TiCDC 软件的常见操作所对应的主要流程进行介绍，帮助用户更好的理解 TiCDC 的工作原理。
```
Comment (IANTHEREAL) [inline ticdc/ticdc-architecture.md:112]:
```suggestion

最后，对 TiCDC 软件的常见操作所对应的主要流程进行介绍，帮助用户更好的理解 TiCDC 的工作原理。
```

File: ticdc/ticdc-architecture.md [hunk 17]
```diff
@@ -0,0 +1,140 @@
...
+对于 owner 的 TiCDC 节点：
+
+  1. 启动 Capture 进程
+  2. 当选 Owner 并启动对应的线程
+  3. 读取 Changefeed 信息
+  4. 启动 Changefeed 管理逻辑
+  5. 根据 Changefeed 配置和最新 checkpointTS, 读取 TiKV 中的 schema 信息，确定需要被同步的表
+  6. 读取各 Processor 当前同步的表的列表，分发需要添加的表
+  7. 更新进度信息
+
```
Comment (IANTHEREAL) [inline ticdc/ticdc-architecture.md:132]:
```suggestion
```

File: ticdc/ticdc-manage-changefeed.md [hunk 18]
```diff
@@ -0,0 +1,309 @@
...
+# 管理 Changefeed
+
+本文介绍 Changefeed 相关的各种管理方法，其中大部分功能是通过 TiCDC 的命令行工具来完成的。如果你需要，也可以通过 TiCDC 暴露的 HTTP 接口实现类似功能，详细信息参考 [TiCDC OpenAPI](/ticdc/ticdc-open-api.md)。
+
+## 创建同步任务
+
+使用以下命令来创建同步任务：
+
+```shell
+cdc cli changefeed create --server=http://10.0.10.25:8300 --sink-uri="mysql://root:123456@127.0.0.1:3306/" --changefeed-id="simple-replication-task" --sort-engine="unified"
```
Comment (IANTHEREAL) [inline ticdc/ticdc-manage-changefeed.md:16]:
```suggestion
cdc cli changefeed create --server=http://10.0.10.25:8300 --sink-uri="mysql://root:123456@127.0.0.1:3306/" --changefeed-id="simple-replication-task"
```

**Review by shichun-0415 — APPROVED** [review]
LGTM

**Comment by IANTHEREAL** [comment]
@shichun-0415 can we merge it?

**Comment by IANTHEREAL** [comment]
LGTM

**Pull request merged**